src/checker/reqwest_checker.rs

```diff
@@ -229,13 +230,58 @@ impl Checker for ReqwestChecker {
     /// Makes a request to a url to determine whether it is up.
     /// Up is defined as returning a 2xx within a specific timeframe.
     #[tracing::instrument]
-    async fn check_url(&self, check: &ScheduledCheck, region: &'static str) -> CheckResult {
+    async fn check_url(
+        &self,
+        check: &ScheduledCheck,
+        region: &'static str,
+        check_robots: bool,
+    ) -> CheckResult {
         let scheduled_check_time = check.get_tick().time();
         let actual_check_time = Utc::now();
         let span_id = SpanId::default();
         let trace_id = make_trace_id(check.get_config(), check.get_tick(), check.get_retry());
         let trace_header = make_trace_header(check.get_config(), &trace_id, span_id);
 
+        if check_robots {
+            if let Ok(url) = check.get_config().url.parse::<Url>() {
+                let mut robots_url = url.clone();
+                robots_url.set_path("robots.txt");
+                let robots_txt = {
+                    let res = self.client.get(robots_url).send().await;
+                    if let Ok(res) = res {
+                        res.bytes().await.unwrap_or_default()
+                    } else {
+                        tracing::debug!("could not retrieve robots.txt");
+                        Bytes::default()
+                    }
+                };
+
+                if let Ok(r) = Robot::new("SentryUptimeBot", &robots_txt) {
+                    if !r.allowed(url.as_str()) {
+                        return CheckResult {
```

Comment:
Hmm, doesn't this mean we're actually skipping a normal check then though? So when we check their robots once a day we won't have checked their configured URL and they'll have a gap there?

I think I would actually expect the way robots should be checked is that the Checker interface gains a new `check_robots` method, which also takes a check, but explicitly only does the robots check.

Then when we're scheduling checks we'll decide if it's time to also execute the robots check and receive those results and produce them